granulate_utils/metadata/databricks_client.py

```diff
@@ -34,55 +36,95 @@ def __init__(self, logger: logging.LoggerAdapter) -> None:
 
-    def _get_job_name_impl(self) -> Optional[str]:
-        # Make sure we're running on a databricks machine
-        if not os.path.isfile(DATABRICKS_DEPLOY_CONF_PATH):
+    @staticmethod
+    def _get_name_from_metadata(metadata: Metadata) -> Optional[str]:
+        if "RunName" in metadata:
+            return str(metadata["RunName"]).replace(" ", "-").lower()
+        return None
+
+    def _cluster_all_tags_metadata(self) -> Optional[Metadata]:
+        """
+        Returns `includes spark.databricks.clusterUsageTags.clusterAllTags` tags as `Dict`.
+        """
+        if not os.path.isfile(DATABRICKS_METRICS_PROP_PATH):
+            # We want to retry in case the cluster is still initializing, and the file is not yet deployed.
             return None
         webui = self.get_webui_address()
         # The API used: https://spark.apache.org/docs/latest/monitoring.html#rest-api
         apps_url = SPARKUI_APPS_URL.format(webui)
-        self.logger.debug(f"Databricks SparkUI address: {apps_url}.")
-        resp = requests.get(apps_url, timeout=REQUEST_TIMEOUT)
-        if not resp.ok:
-            self.logger.warning(
-                f"Failed initializing Databricks client. {apps_url!r} request failed, status_code: {resp.status_code}."
-            )
+        self.logger.debug(f"Databricks SparkUI address: {apps_url}")
+        try:
+            response = self._request_get(apps_url)
+        except requests.exceptions.RequestException:
+            # Request might fail in cases where the cluster is still initializing, retrying.
             return None
-        apps = resp.json()
+        try:
+            apps = response.json()
+        except Exception as e:
+            raise DatabricksJobNameDiscoverException(
+                f"Failed to parse apps url response, query {response.text=}"
+            ) from e
         if len(apps) == 0:
-            self.logger.warning("Failed initializing Databricks client. There are no apps.")
+            # apps might be empty because of initialization, retrying.
+            self.logger.debug("No apps yet, retrying.")
             return None
-        # There's an assumption that only one app exists, and even if there are more -
-        # the name of the job should be the same.
+
         env_url = f"{apps_url}/{apps[0]['id']}/environment"
-        resp = requests.get(env_url, timeout=REQUEST_TIMEOUT)
-        if not resp.ok:
-            self.logger.warning(
-                f"Failed initializing Databricks client. {env_url!r} request failed, status_code: {resp.status_code}."
-            )
-            return None
-        env = resp.json()
-        props = env["sparkProperties"]
+        try:
+            response = self._request_get(env_url)
+        except Exception as e:
+            # No reason for any exception, `environment` uri should be accessible if we have running apps.
+            raise DatabricksJobNameDiscoverException(f"Environment request failed {env_url=}") from e
```

Comment:
It would be easier, later on, to read through these logs, if the values are not formatted into messages but instead added as logging extras. I'm not sure of a proper way to write it, please consider doing it, if you have any idea how to write it nicely.

Reply:
talked f2f, currently no nicer solution